### PR TITLE
This should fix a path enumeration issue

### PR DESCRIPTION
**Pull request by dxa4481**
I got rid of normilize [here](https://github.com/dxa4481/send/blob/master/index.js#L420) because join normilizes, and I added [this](https://github.com/dxa4481/send/blob/master/index.js#L423) to resolve the issue found here https://github.com/TeamMentor/TM_4_0_Design/issues/206

### CODE REVIEW

**Comment by dougwilson** [comment]
Great! I can look into why the changes are making the tests fail.

**Comment by dougwilson** [comment]
The change also seems to make pretty much every single test fail on Windows.

**Comment by dougwilson** [comment]
> The change also seems to make pretty much every single test fail on Windows.

Looks like the main reason for the Windows failures is because `join(root, path)` will never equal `root + path` under any circumstance because of the fact that URLs have forward slashes and Windows uses backslashes in file paths.

**Comment by dxa4481** [comment]
Ahh, gotcha. Sorry I didn't run the tests; I set up the test suite now and am having another go at it.

**Comment by dougwilson** [comment]
No problem :) I just haven't yet gotten to a machine to be able to look into anything yet. Your effort is greatly appreciated!

**Comment by dougwilson** [comment]
Oh, and I think if I were to give the fix a goal, it would be that "at no point, should the path ever descend below the `root`".

**Comment by dougwilson** [comment]
Ok, I got back and I have a perfect solution :)! Basically, just normalize the rhs first, then there is a simple check for a leading up directive.

**Comment by dxa4481** [comment]
Oh man! That's probably simpler than what I ended up doing, but the tests pass with my latest commit

File: index.js [hunk 1]
```diff
@@ -416,9 +416,22 @@ SendStream.prototype.pipe = function(res){
   var parts
   if (root !== null) {
     // join / normalize from optional root dir
-    path = normalize(join(root, path))
+    var rawPath = path
+    path = join(root, path)
     root = normalize(root + sep)
 
+    // check for exiting current dir
+    var traversalCheck = rawPath.split(sep);
```
Comment (dougwilson) [inline index.js]: This still doesn't work on Windows ;) `sep === '\\'` on there, and `path` will just contain forward slashes.

Comment (dougwilson) [inline index.js]: I think _eventually in the far distant future_ we'll finally have `path.split` in Node.js core, haha.

**Comment by dougwilson** [comment]
Ok, @dxa4481 @DinisCruz if you guys want to confirm the current `master` (`npm install tj/send`) and poke at it, I would love a confirmation :) Then I'll roll it up into Express 4.11.1 and 3.19.1, and whatever else I can get my hands on.

**Comment by DinisCruz** [comment]
Great stuff, I'll take a look and its great to see such a fast fix.

I will try to also add a unit test that checks for the edge cases discovered

**Comment by dougwilson** [comment]
Awesome, @DinisCruz :) The final commit does have _one_ unit test (https://github.com/tj/send/commit/98a5b89982b38e79db684177cf94730ce7fc7aed), but if you want to add more, that would be great :)

**Comment by dougwilson** [comment]
So far `send` has been published as 0.11.1
